src/connectedk8s/azext_connectedk8s/_utils.py

```diff
@@ -433,13 +525,360 @@ def can_create_clusterrolebindings(configuration):
+    if not which('kubectl'):
+        raise ValidationError('Can not find kubectl executable in PATH')
+    subprocess_cmd = kubectl_prior + ["get", "node", "--no-headers"]
+    nodes = subprocess.check_output(
+        subprocess_cmd,
+        universal_newlines=True)
+    logger.debug(nodes)
+    node_lines = nodes.splitlines()
+    ready_nodes = {}
+    for node_line in node_lines:
+        columns = node_line.split()
+        logger.debug(node_line)
+        if columns[1] != "Ready":
+            logger.warning("Node %s is not Ready. Current state is: %s.", columns[0], columns[1])
+        else:
+            ready_nodes[columns[0]] = False
+    logger.debug('There are %s ready nodes in the cluster', str(len(ready_nodes)))
+    if not ready_nodes:
+        logger.warning('No nodes are ready in the current cluster. Diagnostics info might not be available.')
+
+    subprocess_cmd = kubectl_prior + ["get", "pods", "-n", "aks-periscope", "--no-headers"]
+    pods = subprocess.check_output(
+        subprocess_cmd,
+        universal_newlines=True)
+    logger.debug(pods)
+    pod_lines = pods.splitlines()
+    running_pods = {}
+    for pod_line in pod_lines:
+        columns = pod_line.split()
+        logger.debug(pod_line)
+        if columns[2] != "Running":
+            logger.warning("Pod %s is not in running state. Current state is %s.", columns[0], columns[2])
+        else:
+            running_pods[columns[0]] = False
+    logger.debug('There are %s ready nodes in the cluster', str(len(ready_nodes)))
+    if not running_pods:
+        logger.warning('We were unable to install diagnostic resources. A possible cause could be lack of resources on your cluster.')
+        return
+
+    network_config_array = []
+    network_status_array = []
+    apds_created = False
+    apd_lines = []
+    max_retry = 10
+    for retry in range(0, max_retry):
+        if not apds_created:
+            subprocess_cmd = kubectl_prior + ["get", "apd", "-n", "aks-periscope", "--no-headers"]
+            try:
+                apd = subprocess.check_output(subprocess_cmd, universal_newlines=True)
+            except subprocess.CalledProcessError as ex:
+                logger.debug(f"Exception while running {subprocess_cmd}: {ex.returncode}, {ex.output}. Retrying...")
+                continue
+
+            apd_lines = apd.splitlines()
+            if apd_lines and 'No resources found' in apd_lines[0]:
+                apd_lines.pop(0)
+
+            print("Got {} diagnostic results for {} ready nodes{}\r".format(len(apd_lines),
+                                                                            len(ready_nodes),
+                                                                            '.' * retry), end='')
+            if len(apd_lines) < len(ready_nodes):
+                logger.warning("Warning: There are %s apd resources, but there are %s nodes running on the cluster."
```

Comment:
User might not understand what an "apd" resource is right? @sophsoph321 Should we mention diagnostic resource instead ?

@shashankbarsin / @mallika-trivedi to suggest too